Userland/Libraries/LibWeb/URL/URL.cpp

```diff
@@ -535,6 +537,29 @@ bool host_is_domain(AK::URL::Host const& host)
     return host.has<String>() && host.get<String>() != String {};
 }
 
+// https://url.spec.whatwg.org/#potentially-strip-trailing-spaces-from-an-opaque-path
+void strip_trailing_spaces_from_an_opaque_path(URL& url)
+{
+    // 1. If url’s URL does not have an opaque path, then return.
+    // FIXME: Reimplement this step once we modernize the URL implementation to the meet spec.
+    if (url.cannot_be_a_base_url())
```

Comment:
Cannot be a base URL is in my understanding equivalent to an opaque path. So this check seems inverted to me. For reference, the web platform tests for this spec change don't match up with the test which is being added here: https://github.com/web-platform-tests/wpt/pull/37556/files

Reply:
Thanks for spotting this. From looking at the tests you provided and scouring the specs, I found that [special URL paths](https://url.spec.whatwg.org/#concept-url-path) cannot be an opaque path (see the `Note` section beneath the link).  The changes I made to the JS tests should reflect this.